rfcs/rfc-006-budget-contract-language.md

```diff
@@ -0,0 +1,122 @@
+
+### Wallet CLI
+
+The general form is:
+```
+$ solana-wallet [common-options] [command] [command-specific options]
+```
+`common-options` include:
+* `--fee xyz` - Transaction fee (0 by default)
+* `--output file` - Write the raw Transaction to a file instead of sending it
+
+`command` variants:
+* `pay`
+* `cancel`
+* `apply`
+
+#### Unconditional Immediate Transfer
+```sh
+// Command
+$ solana-wallet pay RecipientPubkey 123
+
+// Return
+TxSignature
+```
+
+#### Time-based Transfer
+```sh
+// Command
+$ solana-wallet pay RecipientPubkey 123 \
+    --after 2018-12-24T23:59 --from TimestampPubKey
+
+// Return
+{signature: TxSignature, contractId: ContractId}
```

Comment:
Have there been any thoughts on re-using (sharing) contract code?  If so that's another argument for separating out the contract id from the process id.

Reply:
Yes, exactly.